src/main/kotlin/com/learning/medicare/user/UserService.kt

```diff
@@ -41,7 +41,6 @@ interface UserServiceContract {
     fun findOne(id: Long): User
     fun save(user: User): User
 
-    @Throws(UserNotFoundException::class, InvalidAssociationException::class, InvalidCaregiver::class)
-    fun associate(patientId: Long, caregiverId: Long): TakesCareOf
+    fun associate(patientId: Long, caregiverId: Long): Result<TakesCareOf, Exception>
```

Comment:
I removed the annotations for checked exceptions (some tests are failing because of this), and I was wondering if having `Exception` as a 2nd args for the Result type is correct. My thought was that since Kotlin only has runtime exceptions, `Result` uses the 2nd arg to differentiate if a method can throw an exception or not (in their docs if a method doesn't, `NoException` can be used).